### PR TITLE
Fix settings won't sync on core v20.8.1

### LINKED ISSUE #27: Overwrites existing pianobar config
I already use pianobar, but I feed it my password via my password manager. I was able to modify this for my personal use to not mess with my config and not require my password in plaintext, but I feel like there should really be a fix for this here (which is a separate issue I'm going to report shortly).

When a pianobar config already exists, it shouldn't be overwritten. Appending the event handler, sure, but that can be done without overwriting other custom settings.

### PR DESCRIPTION
**Pull request by krisgesling**
As reported in https://github.com/MycroftAI/mycroft-core/issues/2842

The Skill settings stopped syncing due to the Skill storing information on currently and recently played items in the Skill settings object.

This moves all values that are not Skill settings to a local json database stored in the Skills namespaced filesystem.

An auto-format and small import clean up was also done. 

FuzzyWuzzy has been updated to incorporate [fixes for Python3.7](https://github.com/seatgeek/fuzzywuzzy/releases/tag/0.18.0).

### To Test
- Ensure you are running mycroft-core 20.8.1 (or latest dev branch)
- Install new version of Skill - if switching branches ensure you get the new dependencies
  ```
  mycroft-pip install -r /opt/mycroft/skills/mycroft-pandora.mycroftai/requirements.txt
  ```
- Delete any existing settings.json
- Reboot Mycroft.

### CODE REVIEW

**Comment by thejacer** [comment]
[skills.log](https://github.com/MycroftAI/pianobar-skill/files/6001044/skills.log)

Same result. I'm sorry, I don't know how best to provide the skills.log file. I made sure to use branch bugfix/settings-sync and deleted my settings.json file prior to rebooting the machine and testing.

2021-02-18 07:41:05.748 | INFO     |  7139 | mycroft.messagebus.load_config:load_message_bus_config:33 | Loading message bus configs
2021-02-18 07:41:05.761 | INFO     |  7139 | mycroft.messagebus.client.client:on_open:114 | Connected
2021-02-18 07:41:05.764 | INFO     |  7139 | mycroft.util.process_utils:start_message_bus_client:155 | Connected to messagebus
2021-02-18 07:41:05.772 | INFO     |  7139 | mycroft.skills.msm_wrapper:create_msm:95 | Acquiring lock to instantiate MSM
2021-02-18 07:41:05.775 | INFO     |  7139 | msm.mycroft_skills_manager | building SkillEntry objects for all skills
2021-02-18 07:41:08.985 | INFO     |  7139 | mycroft.skills.msm_wrapper:create_msm:111 | Releasing MSM instantiation lock.
2021-02-18 07:41:08.987 | INFO     |  7139 | mycroft.skills.skill_updater:_log_next_download_time:265 | Next scheduled skill update: 2021-02-18 07:48:50.575659
2021-02-18 07:41:08.990 | INFO     |  7139 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: mycroft-pairing.mycroftai
2021-02-18 07:41:09.005 | INFO     |  7139 | mycroft.skills.settings:get_local_settings:83 | /home/pi/.config/mycroft/skills/mycroft-pairing.mycroftai/settings.json
2021-02-18 07:41:09.037 | INFO     |  7139 | mycroft.skills.skill_loader:_communicate_load_status:344 | Skill mycroft-pairing.mycroftai loaded successfully
2021-02-18 07:41:09.039 | INFO     |  7139 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: mycroft-volume.mycroftai
2021-02-18 07:41:09.065 | INFO     |  7139 | mycroft.skills.settings:get_local_settings:83 | /home/pi/.config/mycroft/skills/mycroft-volume.mycroftai/settings.json
2021-02-18 07:41:09.376 | INFO     |  7139 | mycroft.skills.skill_loader:_communicate_load_status:344 | Skill mycroft-volume.mycroftai loaded successfully
2021-02-18 07:41:09.381 | INFO     |  7139 | __main__:on_started:175 | Skills service is starting up.
2021-02-18 07:41:09.550 | INFO     |  7139 | __main__:_get_pairing_status:90 | Device is paired
2021-02-18 07:41:09.573 | INFO     |  7139 | __main__:_update_device_attributes_on_backend:142 | Sending updated device attributes to the backend...
2021-02-18 07:41:09.898 | INFO     |  7139 | __main__:on_alive:179 | Skills service is alive.
2021-02-18 07:41:09.992 | INFO     |  7139 | mycroft.skills.skill_manager:_load_on_startup:269 | Loading installed skills...
2021-02-18 07:41:09.997 | INFO     |  7139 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: mycroft-personal.mycroftai
2021-02-18 07:41:10.006 | INFO     |  7139 | mycroft.skills.settings:get_local_settings:83 | /home/pi/.config/mycroft/skills/mycroft-personal.mycroftai/settings.json
2021-02-18 07:41:10.091 | INFO     |  7139 | mycroft.skills.skill_loader:_communicate_load_status:344 | Skill mycroft-personal.mycroftai loaded successfully
2021-02-18 07:41:10.094 | INFO     |  7139 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: mycroft-reminder.mycroftai
2021-02-18 07:41:10.114 | INFO     |  7139 | mycroft.skills.settings:get_local_settings:83 | /home/pi/.config/mycroft/skills/mycroft-reminder.mycroftai/settings.json
2021-02-18 07:41:10.259 | INFO     |  7139 | mycroft.skills.skill_loader:_communicate_load_status:344 | Skill mycroft-reminder.mycroftai loaded successfully
2021-02-18 07:41:10.266 | INFO     |  7139 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: mycroft-timer.mycroftai
2021-02-18 07:41:10.353 | INFO     |  7139 | mycroft.skills.settings:get_local_settings:83 | /home/pi/.config/mycroft/skills/mycroft-timer.mycroftai/settings.json
2021-02-18 07:41:10.538 | INFO     |  7139 | mycroft.skills.skill_loader:_communicate_load_status:344 | Skill mycroft-timer.mycroftai loaded successfully
2021-02-18 07:41:10.540 | INFO     |  7139 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: mycroft-wiki.mycroftai
2021-02-18 07:41:10.876 | INFO     |  7139 | mycroft.skills.settings:get_local_settings:83 | /home/pi/.config/mycroft/skills/mycroft-wiki.mycroftai/settings.json
2021-02-18 07:41:10.926 | INFO     |  7139 | mycroft.skills.skill_loader:_communicate_load_status:344 | Skill mycroft-wiki.mycroftai loaded successfully
2021-02-18 07:41:10.930 | INFO     |  7139 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: mycroft-stop.mycroftai
2021-02-18 07:41:10.948 | INFO     |  7139 | mycroft.skills.settings:get_local_settings:83 | /home/pi/.config/mycroft/skills/mycroft-stop.mycroftai/settings.json
2021-02-18 07:41:11.034 | INFO     |  7139 | mycroft.skills.skill_loader:_communicate_load_status:344 | Skill mycroft-stop.mycroftai loaded successfully
2021-02-18 07:41:11.038 | INFO     |  7139 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: mycroft-naptime.mycroftai
2021-02-18 07:41:11.061 | INFO     |  7139 | mycroft.skills.settings:get_local_settings:83 | /home/pi/.config/mycroft/skills/mycroft-naptime.mycroftai/settings.json
2021-02-18 07:41:11.095 | INFO     |  7139 | mycroft.skills.skill_loader:_communicate_load_status:344 | Skill mycroft-naptime.mycroftai loaded successfully
2021-02-18 07:41:11.098 | INFO     |  7139 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: mycroft-spelling.mycroftai
2021-02-18 07:41:11.113 | INFO     |  7139 | mycroft.skills.settings:get_local_settings:83 | /home/pi/.config/mycroft/skills/mycroft-spelling.mycroftai/settings.json
2021-02-18 07:41:11.150 | INFO     |  7139 | mycroft.skills.skill_loader:_communicate_load_status:344 | Skill mycroft-spelling.mycroftai loaded successfully
2021-02-18 07:41:11.155 | INFO     |  7139 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: mycroft-stock.mycroftai
2021-02-18 07:41:11.172 | INFO     |  7139 | mycroft.skills.settings:get_local_settings:83 | /opt/mycroft/skills/mycroft-stock.mycroftai/settings.json
2021-02-18 07:41:11.173 | INFO     |  7139 | StockSkill | The Stock Skill has been disabled due to a
2021-02-18 07:41:11.173 | INFO     |  7139 | StockSkill | breaking change made to the 3rd party API
2021-02-18 07:41:11.173 | INFO     |  7139 | StockSkill | For further information, see:
2021-02-18 07:41:11.173 | INFO     |  7139 | StockSkill | https://github.com/MycroftAI/skill-stock/issues/31
2021-02-18 07:41:11.175 | ERROR    |  7139 | mycroft.skills.skill_loader:_create_skill_instance:295 | Skill __init__ failed with Exception('Skill has been intentionally disabled by Mycroft')
Traceback (most recent call last):
  File "/home/pi/mycroft-core/mycroft/skills/skill_loader.py", line 292, in _create_skill_instance
    self.instance = skill_module.create_skill()
  File "/opt/mycroft/skills/mycroft-stock.mycroftai/__init__.py", line 131, in create_skill
    return StockSkill()
  File "/opt/mycroft/skills/mycroft-stock.mycroftai/__init__.py", line 91, in __init__
    raise Exception('Skill has been intentionally disabled by Mycroft')
Exception: Skill has been intentionally disabled by Mycroft
2021-02-18 07:41:11.179 | ERROR    |  7139 | mycroft.skills.skill_loader:_communicate_load_status:351 | Skill mycroft-stock.mycroftai failed to load
2021-02-18 07:41:11.182 | INFO     |  7139 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: fallback-aiml.forslund
2021-02-18 07:41:11.210 | INFO     |  7139 | mycroft.skills.settings:get_local_settings:83 | /home/pi/.config/mycroft/skills/fallback-aiml.forslund/settings.json
2021-02-18 07:41:11.244 | INFO     |  7139 | mycroft.skills.skill_loader:_communicate_load_status:344 | Skill fallback-aiml.forslund loaded successfully
2021-02-18 07:41:11.246 | INFO     |  7139 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: mycroft-audio-record.mycroftai
2021-02-18 07:41:11.263 | INFO     |  7139 | mycroft.skills.settings:get_local_settings:83 | /home/pi/.config/mycroft/skills/mycroft-audio-record.mycroftai/settings.json
Removing event mycroft-timer.mycroftai:ShowTimer
2021-02-18 07:41:11.319 | INFO     |  7139 | mycroft.skills.skill_loader:_communicate_load_status:344 | Skill mycroft-audio-record.mycroftai loaded successfully
2021-02-18 07:41:11.322 | INFO     |  7139 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: skill-finished-booting.zelmon64
2021-02-18 07:41:11.333 | INFO     |  7139 | mycroft.skills.settings:get_local_settings:83 | /home/pi/.config/mycroft/skills/skill-finished-booting.zelmon64/settings.json
2021-02-18 07:41:11.364 | INFO     |  7139 | mycroft.skills.skill_loader:_communicate_load_status:344 | Skill skill-finished-booting.zelmon64 loaded successfully
2021-02-18 07:41:11.366 | INFO     |  7139 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: mycroft-alarm.mycroftai
2021-02-18 07:41:11.383 | INFO     |  7139 | mycroft.skills.settings:get_local_settings:83 | /home/pi/.config/mycroft/skills/mycroft-alarm.mycroftai/settings.json
Removing event mycroft-alarm.mycroftai:NextAlarm
2021-02-18 07:41:11.569 | INFO     |  7139 | mycroft.skills.skill_loader:_communicate_load_status:344 | Skill mycroft-alarm.mycroftai loaded successfully
2021-02-18 07:41:11.573 | INFO     |  7139 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: mycroft-support-helper.mycroftai
2021-02-18 07:41:11.599 | INFO     |  7139 | mycroft.skills.settings:get_local_settings:83 | /home/pi/.config/mycroft/skills/mycroft-support-helper.mycroftai/settings.json
2021-02-18 07:41:11.636 | INFO     |  7139 | mycroft.skills.skill_loader:_communicate_load_status:344 | Skill mycroft-support-helper.mycroftai loaded successfully
2021-02-18 07:41:11.639 | INFO     |  7139 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: mycroft-date-time.mycroftai
2021-02-18 07:41:12.161 | INFO     |  7139 | mycroft.skills.settings:get_local_settings:83 | /home/pi/.config/mycroft/skills/mycroft-date-time.mycroftai/settings.json
2021-02-18 07:41:12.290 | INFO     |  7139 | TimeSkill | Registering resting screen <bound method TimeSkill.handle_idle of <mycroft-date-time_mycroftai.TimeSkill object at 0x85a62f10>> for Time and Date.
2021-02-18 07:41:12.298 | INFO     |  7139 | TimeSkill | Registering resting screen
2021-02-18 07:41:12.310 | INFO     |  7139 | mycroft.skills.skill_loader:_communicate_load_status:344 | Skill mycroft-date-time.mycroftai loaded successfully
2021-02-18 07:41:12.314 | INFO     |  7139 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: mycroft-version-checker.mycroftai
2021-02-18 07:41:12.330 | INFO     |  7139 | mycroft.skills.settings:get_local_settings:83 | /home/pi/.config/mycroft/skills/mycroft-version-checker.mycroftai/settings.json
2021-02-18 07:41:12.402 | INFO     |  7139 | mycroft.skills.skill_loader:_communicate_load_status:344 | Skill mycroft-version-checker.mycroftai loaded successfully
2021-02-18 07:41:12.404 | INFO     |  7139 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: mycroft-configuration.mycroftai
2021-02-18 07:41:12.416 | INFO     |  7139 | mycroft.skills.settings:get_local_settings:83 | /home/pi/.config/mycroft/skills/mycroft-configuration.mycroftai/settings.json
2021-02-18 07:41:12.661 | INFO     |  7139 | mycroft.skills.skill_loader:_communicate_load_status:344 | Skill mycroft-configuration.mycroftai loaded successfully
2021-02-18 07:41:12.663 | INFO     |  7139 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: mycroft-npr-news.mycroftai
2021-02-18 07:41:12.822 | INFO     |  7139 | mycroft.skills.settings:get_local_settings:83 | /home/pi/.config/mycroft/skills/mycroft-npr-news.mycroftai/settings.json
2021-02-18 07:41:13.882 | INFO     |  7139 | mycroft.skills.skill_loader:_communicate_load_status:344 | Skill mycroft-npr-news.mycroftai loaded successfully
2021-02-18 07:41:13.884 | INFO     |  7139 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: mycroft-hello-world.mycroftai
2021-02-18 07:41:13.900 | INFO     |  7139 | mycroft.skills.settings:get_local_settings:83 | /home/pi/.config/mycroft/skills/mycroft-hello-world.mycroftai/settings.json
2021-02-18 07:41:13.949 | INFO     |  7139 | mycroft.skills.skill_loader:_communicate_load_status:344 | Skill mycroft-hello-world.mycroftai loaded successfully
2021-02-18 07:41:13.954 | INFO     |  7139 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: mycroft-singing.mycroftai
2021-02-18 07:41:13.978 | INFO     |  7139 | mycroft.skills.settings:get_local_settings:83 | /home/pi/.config/mycroft/skills/mycroft-singing.mycroftai/settings.json
2021-02-18 07:41:14.020 | INFO     |  7139 | mycroft.skills.skill_loader:_communicate_load_status:344 | Skill mycroft-singing.mycroftai loaded successfully
2021-02-18 07:41:14.027 | INFO     |  7139 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: mycroft-pandora.mycroftai
2021-02-18 07:41:14.055 | INFO     |  7139 | mycroft.skills.settings:get_local_settings:83 | /opt/mycroft/skills/mycroft-pandora.mycroftai/settings.json
pianobar: no process found
2021-02-18 07:41:14.246 | INFO     |  7139 | mycroft.skills.skill_loader:_check_for_first_run:324 | First run of mycroft-pandora.mycroftai
2021-02-18 07:41:14.253 | INFO     |  7139 | mycroft.skills.settings:save_settings:116 | Skill settings successfully saved to /home/pi/.config/mycroft/skills/mycroft-pandora.mycroftai/settings.json
2021-02-18 07:41:14.259 | INFO     |  7139 | mycroft.skills.skill_loader:_communicate_load_status:344 | Skill mycroft-pandora.mycroftai loaded successfully
2021-02-18 07:41:14.263 | INFO     |  7139 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: mycroft-speak.mycroftai
2021-02-18 07:41:14.307 | INFO     |  7139 | mycroft.skills.settings:get_local_settings:83 | /home/pi/.config/mycroft/skills/mycroft-speak.mycroftai/settings.json
2021-02-18 07:41:14.415 | INFO     |  7139 | mycroft.skills.skill_loader:_communicate_load_status:344 | Skill mycroft-speak.mycroftai loaded successfully
2021-02-18 07:41:14.420 | INFO     |  7139 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: mycroft-playback-control.mycroftai
2021-02-18 07:41:14.434 | INFO     |  7139 | mycroft.skills.settings:get_local_settings:83 | /home/pi/.config/mycroft/skills/mycroft-playback-control.mycroftai/settings.json
2021-02-18 07:41:14.580 | INFO     |  7139 | mycroft.skills.skill_loader:_communicate_load_status:344 | Skill mycroft-playback-control.mycroftai loaded successfully
2021-02-18 07:41:14.582 | INFO     |  7139 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: mycroft-ip.mycroftai
2021-02-18 07:41:14.682 | INFO     |  7139 | mycroft.skills.settings:get_local_settings:83 | /home/pi/.config/mycroft/skills/mycroft-ip.mycroftai/settings.json
2021-02-18 07:41:14.754 | INFO     |  7139 | mycroft.skills.skill_loader:_communicate_load_status:344 | Skill mycroft-ip.mycroftai loaded successfully
2021-02-18 07:41:14.759 | INFO     |  7139 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: respeaker-4mic-hat-skill.j1nx
2021-02-18 07:41:14.937 | INFO     |  7139 | mycroft.skills.settings:get_local_settings:83 | /home/pi/.config/mycroft/skills/respeaker-4mic-hat-skill.j1nx/settings.json
2021-02-18 07:41:14.965 | INFO     |  7139 | ReSpeaker_4mic_hat | Pixel Ring: Initializing
2021-02-18 07:41:14.981 | INFO     |  7139 | ReSpeaker_4mic_hat | Pixel Ring: Enabling
2021-02-18 07:41:15.002 | INFO     |  7139 | mycroft.skills.skill_loader:_communicate_load_status:344 | Skill respeaker-4mic-hat-skill.j1nx loaded successfully
2021-02-18 07:41:15.004 | INFO     |  7139 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: fallback-unknown.mycroftai
2021-02-18 07:41:15.017 | INFO     |  7139 | mycroft.skills.settings:get_local_settings:83 | /home/pi/.config/mycroft/skills/fallback-unknown.mycroftai/settings.json
2021-02-18 07:41:15.050 | INFO     |  7139 | mycroft.skills.skill_loader:_communicate_load_status:344 | Skill fallback-unknown.mycroftai loaded successfully
2021-02-18 07:41:15.052 | INFO     |  7139 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: fallback-wolfram-alpha.mycroftai
2021-02-18 07:41:15.097 | INFO     |  7139 | mycroft.skills.settings:get_local_settings:83 | /home/pi/.config/mycroft/skills/fallback-wolfram-alpha.mycroftai/settings.json
2021-02-18 07:41:15.167 | INFO     |  7139 | mycroft.skills.skill_loader:_communicate_load_status:344 | Skill fallback-wolfram-alpha.mycroftai loaded successfully
2021-02-18 07:41:15.173 | INFO     |  7139 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: mycroft-weather.mycroftai
2021-02-18 07:41:15.673 | INFO     |  7139 | mycroft.skills.settings:get_local_settings:83 | /home/pi/.config/mycroft/skills/mycroft-weather.mycroftai/settings.json
Removing event mycroft-weather.mycroftai:precache1
Removing event mycroft-weather.mycroftai:precache2
Removing event mycroft-weather.mycroftai:precache3
2021-02-18 07:41:16.825 | INFO     |  7139 | mycroft.skills.skill_loader:_communicate_load_status:344 | Skill mycroft-weather.mycroftai loaded successfully
2021-02-18 07:41:16.830 | INFO     |  7139 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: mycroft-installer.mycroftai
2021-02-18 07:41:16.855 | INFO     |  7139 | mycroft.skills.settings:get_local_settings:83 | /home/pi/.config/mycroft/skills/mycroft-installer.mycroftai/settings.json
2021-02-18 07:41:16.993 | INFO     |  7139 | mycroft.skills.skill_loader:_communicate_load_status:344 | Skill mycroft-installer.mycroftai loaded successfully
2021-02-18 07:41:16.994 | INFO     |  7139 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: mycroft-joke.mycroftai
2021-02-18 07:41:17.021 | INFO     |  7139 | mycroft.skills.settings:get_local_settings:83 | /home/pi/.config/mycroft/skills/mycroft-joke.mycroftai/settings.json
2021-02-18 07:41:17.060 | INFO     |  7139 | mycroft.skills.skill_loader:_communicate_load_status:344 | Skill mycroft-joke.mycroftai loaded successfully
2021-02-18 07:41:17.064 | INFO     |  7139 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: mycroft-fallback-duck-duck-go.mycroftai
2021-02-18 07:41:17.105 | INFO     |  7139 | mycroft.skills.settings:get_local_settings:83 | /home/pi/.config/mycroft/skills/mycroft-fallback-duck-duck-go.mycroftai/settings.json
2021-02-18 07:41:17.140 | INFO     |  7139 | mycroft.skills.skill_loader:_communicate_load_status:344 | Skill mycroft-fallback-duck-duck-go.mycroftai loaded successfully
2021-02-18 07:41:17.143 | INFO     |  7139 | mycroft.skills.skill_loader:load:185 | ATTEMPTING TO LOAD SKILL: fallback-query.mycroftai
2021-02-18 07:41:17.155 | INFO     |  7139 | mycroft.skills.settings:get_local_settings:83 | /home/pi/.config/mycroft/skills/fallback-query.mycroftai/settings.json
2021-02-18 07:41:17.176 | INFO     |  7139 | mycroft.skills.skill_loader:_communicate_load_status:344 | Skill fallback-query.mycroftai loaded successfully
2021-02-18 07:41:17.179 | INFO     |  7139 | mycroft.skills.skill_manager:_load_on_startup:271 | Skills all loaded!
2021-02-18 07:41:17.184 | INFO     |  7139 | mycroft.skills.skill_manager:_start_settings_update:190 | Start settings update
2021-02-18 07:41:17.184 | INFO     |  7139 | msm.mycroft_skills_manager | invalidating skills cache
2021-02-18 07:41:17.185 | INFO     |  7139 | msm.mycroft_skills_manager | building SkillEntry objects for all skills
2021-02-18 07:41:17.234 | INFO     |  7139 | mycroft.skills.intent_services.padatious_service:train:81 | Training... (single_thread=False)
2021-02-18 07:41:17.243 | INFO     |  7139 | __main__:on_ready:183 | Skills service is ready.
/home/pi/mycroft-core/.venv/lib/python3.7/site-packages/fuzzywuzzy/fuzz.py:11: UserWarning: Using slow pure-python SequenceMatcher. Install python-Levenshtein to remove this warning
  warnings.warn('Using slow pure-python SequenceMatcher. Install python-Levenshtein to remove this warning')
/home/pi/mycroft-core/.venv/lib/python3.7/site-packages/gpiozero/devices.py:279: PinFactoryFallback: Falling back from rpigpio: No module named 'RPi'
  'Falling back from %s: %s' % (name, str(e))))
/home/pi/mycroft-core/.venv/lib/python3.7/site-packages/gpiozero/devices.py:279: PinFactoryFallback: Falling back from rpio: No module named 'RPIO'
  'Falling back from %s: %s' % (name, str(e))))
/home/pi/mycroft-core/.venv/lib/python3.7/site-packages/gpiozero/devices.py:279: PinFactoryFallback: Falling back from pigpio: No module named 'pigpio'
  'Falling back from %s: %s' % (name, str(e))))
2021-02-18 07:41:17.828 | INFO     |  7139 | mycroft.skills.intent_services.padatious_service:train:83 | Training complete.
2021-02-18 07:41:17.897 | INFO     |  7139 | ReSpeaker_4mic_hat | Pixel Ring: Speak
2021-02-18 07:41:19.160 | INFO     |  7139 | ReSpeaker_4mic_hat | Pixel Ring: Off
2021-02-18 07:41:21.404 | INFO     |  7139 | mycroft.skills.skill_manager:send:64 | New Settings meta to upload.
2021-02-18 07:41:24.178 | WARNING  |  7139 | mycroft.skills.settings:_update_settings_meta:298 | DEPRECATION WARNING: The "name" attribute in the settingsmeta file is no longer supported.
2021-02-18 07:41:28.782 | WARNING  |  7139 | mycroft.skills.settings:_update_settings_meta:298 | DEPRECATION WARNING: The "name" attribute in the settingsmeta file is no longer supported.
2021-02-18 07:41:31.191 | INFO     |  7139 | mycroft.skills.skill_manager:_start_settings_update:193 | All settings meta has been processed or upload has started
2021-02-18 07:41:31.510 | INFO     |  7139 | mycroft.skills.skill_manager:_start_settings_update:195 | Skill settings downloading has started
2021-02-18 07:41:59.074 | INFO     |  7139 | ReSpeaker_4mic_hat | Pixel Ring: Wakeup
2021-02-18 07:42:00.828 | INFO     |  7139 | geocoder.base | Requested https://nominatim.openstreetmap.org/search?q=America%2FChicago&format=jsonv2&addressdetails=1&limit=1
2021-02-18 07:42:01.702 | INFO     |  7139 | ReSpeaker_4mic_hat | Pixel Ring: Off
2021-02-18 07:42:02.579 | INFO     |  7139 | Playback Control Skill | Resolving Player for: pandora
2021-02-18 07:42:02.581 | INFO     |  7139 | ReSpeaker_4mic_hat | Pixel Ring: Think
2021-02-18 07:42:02.811 | INFO     |  7139 | ReSpeaker_4mic_hat | Pixel Ring: Speak
2021-02-18 07:42:04.656 | INFO     |  7139 | ReSpeaker_4mic_hat | Pixel Ring: Off
2021-02-18 07:42:04.712 | INFO     |  7139 | ReSpeaker_4mic_hat | Pixel Ring: Off
2021-02-18 07:42:04.714 | INFO     |  7139 | mycroft.skills.settings:save_settings:116 | Skill settings successfully saved to /home/pi/.config/mycroft/skills/mycroft-pandora.mycroftai/settings.json
Removing event mycroft-playback-control.mycroftai:PlayQueryTimeout
Removing event mycroft-playback-control.mycroftai:PlayQueryTimeout
Removing event mycroft-playback-control.mycroftai:PlayQueryTimeout
2021-02-18 07:42:09.404 | INFO     |  7139 | Playback Control Skill | Playing with: mycroft-pandora.mycroftai
2021-02-18 07:42:09.844 | INFO     |  7139 | Playback Control Skill | Audio service status: {}
2021-02-18 07:42:09.929 | INFO     |  7139 | ReSpeaker_4mic_hat | Pixel Ring: Speak
2021-02-18 07:42:12.973 | INFO     |  7139 | ConfigurationSkill | Remote configuration updated
2021-02-18 07:42:13.629 | INFO     |  7139 | ReSpeaker_4mic_hat | Pixel Ring: Off

**Comment by JarbasAl** [comment]
consider using https://github.com/maxbachmann/rapidfuzz

fuzzywuzzy is GPL

File: __init__.py [hunk 1]
```diff
@@ -193,138 +210,146 @@ def on_websettings_changed(self):
...
             self.cmd("0\n")
             self.cmd("S")
             time.sleep(0.5)
             self.process.kill()
-            self.settings['first_init'] = False
+            self.play_info["first_init"] = False
             self._load_current_info()
         except Exception as e:
-            LOG.exception('Failed to connect to Pandora: '+repr(e))
+            self.log.exception("Failed to connect to Pandora: " + repr(e))
```
Comment (chrisveilleux) [inline __init__.py]: You don't need the `repr(e)` in a log.exception as it is automatically included in the log message.

**Review by chrisveilleux — APPROVED** [review]
One small thing on a log exception.  Otherwise, good work!

File: __init__.py (hunk 1 above)
Comment (krisgesling) [reply __init__.py]: Good spot, fixed.

**Pull request merged**